pages/react-canvas-components.mdx

```diff
@@ -0,0 +1,256 @@
+
+```js
+// moving our hand to the starting position
+canvasContext.moveTo(x1, y1);
+// drawing a blueprint line to the finishing position
+canvasContext.lineTo(x2, y2);
+// taking a purple pen and coloring the line
+canvasContext.strokeStyle = 'purple';
+canvasContext.stroke();
+```
+
+Having imperative code like this in a component oriented codebase can be tricky ! Let's create a component that renders a `<canvas />` on the page. Then, we need to call the `moveTo` and `lineTo` methods on it to draw a line. In practice, it's a bit more complicated to bridge those two. We need to use a [React reference](https://thibaut.io/react-refs-evolution) to access the canvas DOM node, then need to get a 2D context out of it, and finally we can call our drawing methods. The code looks like that:
+
+```jsx basic-canvas-hs3cy src/App.js
+const Canvas = () => {
+  // we use a ref to access the canvas' DOM node
+  const canvasRef = React.useRef(null);
+
+  React.useEffect(() => {
+    const context = canvasRef.current.getContext('2d');
+    // ...drawing using the context
+  }, [canvasRef]);
+
+  return <canvas ref={canvasRef} />;
+};
+```
+
+But if we want to draw something a bit more complex, the `Canvas` component can become quite large. Usually, big components are split into several child components. Yet here, this is not possible as there is only one `canvas` node.
+
+## 🎨 Hexagons
+
+To show how to make child components with canvas, let's draw something more fancy ✨
+
+<div
+  style={{
+    height: '360px',
+    width: '100%',
+  }}
+>
+  <RandomHexagons isAnimating={true} />
+</div>
+
+A single hexagon is defined with the folowing data:
+
+- A position on the screen, defined by two `x` and `y` number values.
+- A `radius` that defines its size.
+- A `rotation` so that all hexagons don't look aligned.
+- A `color`.
+
+We need a function that is able to generate some random hexagons to display. The randomisation code is not relevant here, let's assume we have a way to get an array of hexagons. As for how to draw the shape of an hexagon, we need to draw a line between all its corners, and then fill it with a color:
+
+```jsx
+// This article explains all the math behind hexagons
+// https://www.redblobgames.com/grids/hexagons/
+const corners = getHexagonCorners(x, y, radius, rotation);
+
+context.beginPath();
+corners.forEach((corner, index) => {
+  if (index === 0) {
+    context.moveTo(corner.x, corner.y);
+  } else {
+    context.lineTo(corner.x, corner.y);
+  }
+});
+context.fillStyle = color;
+context.fill();
+```
+
+Now, how could we extract this logic into its own `Hexagon` component ? We need the canvas' context to draw anything on it. We could pass it to all the child components via a prop, but this approach can be a bit tedious when children are nested. Another way of doing this is by using a [React context](https://reactjs.org/docs/context.html) to share "global" values between components.
```

Comment:
- Take out "Now" and remove space before exclamation point
- "canvas'" => "canvas's" 
- "The component would need the canvas's context in order to draw anything. This could be passed via a prop to all child components, but this approach can become tedious when children are deeply nested"